### PR TITLE
New demo: peer connection to Web Audio

### PR DESCRIPTION
**Pull request by samdutton**
Tested in Firefox 43 and Chrome 50 on desktop and Android. 

Online at https://samdutton.github.io/webrtc/src/content/peerconnection/webaudio-output.

One thing I don't understand: this demo requires the remote stream to be set as the srcObject of a video element (see [main.js line 160](https://github.com/samdutton/webrtc/blob/d61009c49e9fb594459238e1ef3d574efd6beb2a/src/content/peerconnection/webaudio-output/js/main.js#L160)). 

This means that video#remoteVideo is in the HTML, but hidden. (I don't want to show it, because the layout works better just showing the local video and the canvas — especially on mobile.)

Is there some way around this?

### CODE REVIEW

File: src/content/peerconnection/webaudio-output/js/test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/*
```
Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-output/js/test.js]: Remove this file unless you add a complete test, grunt currently fails on this.

File: src/content/peerconnection/webaudio-output/css/main.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/*
+*  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
```
Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-output/css/main.css]: Update copyright year, applies throughout.

File: src/content/peerconnection/webaudio-output/index.html [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  <meta name="viewport" content="width=device-width, user-scalable=yes, initial-scale=1, maximum-scale=1">
+  <meta itemprop="description" content="Client-side WebRTC code samples">
+  <meta itemprop="image" content="../../../images/webrtc-icon-192x192.png">
+  <meta itemprop="name" content="WebRTC code samples">
+  <meta name="mobile-web-app-capable" content="yes">
+  <meta id="theme-color" name="theme-color" content="#ffffff">
+
+  <base target="_blank">
+
+  <title>Peer connection as input to Web Audio</title>
```
Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-output/index.html:23]: Should this be PeerConnection?

File: src/content/peerconnection/webaudio-output/js/main.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+  trace('Local video videoWidth: ' + this.videoWidth +
+    'px,  videoHeight: ' + this.videoHeight + 'px');
+});
+
+remoteVideo.addEventListener('loadedmetadata', function() {
+  trace('Remote video videoWidth: ' + this.videoWidth +
+    'px,  videoHeight: ' + this.videoHeight + 'px');
+});
+
+remoteVideo.onresize = function() {
```
Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-output/js/main.js]: maybe use addEventListener here as well for consistency as you have above.

File: src/content/peerconnection/webaudio-output/js/main.js [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+function onIceStateChange(pc, event) {
+  if (pc) {
+    trace(getName(pc) + ' ICE state: ' + pc.iceConnectionState);
+    console.log('ICE state change event: ', event);
+  }
+}
+
+function hangup() {
+  trace('Ending call');
```
Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-output/js/main.js:209]: Does the canvas tag need to be "reset" as well between calls?

**Comment by KaptenJansson** [comment]
Nice demo! LGTM except for some nits.

File: src/content/peerconnection/webaudio-output/js/test.js (hunk 1 above)
Comment (samdutton) [inline src/content/peerconnection/webaudio-output/js/test.js]: Done.

File: src/content/peerconnection/webaudio-output/css/main.css (hunk 2 above)
Comment (samdutton) [inline src/content/peerconnection/webaudio-output/css/main.css]: D'oh! Thanks. Done.

File: src/content/peerconnection/webaudio-output/index.html (hunk 3 above)
Comment (samdutton) [inline src/content/peerconnection/webaudio-output/index.html:23]: Not sure. I copied the style from the other Web Audio demo. 

Maybe RTCPeerConnection? Agree it's a little odd.

File: src/content/peerconnection/webaudio-output/js/main.js (hunk 4 above)
Comment (samdutton) [inline src/content/peerconnection/webaudio-output/js/main.js]: Good spot! Done.

File: src/content/peerconnection/webaudio-output/js/main.js (hunk 5 above)
Comment (samdutton) [inline src/content/peerconnection/webaudio-output/js/main.js:209]: I could clear the canvas in `hangup()`, but I don't think that's necessary. 

Also noticed that repeated calls each create a new AudioContext, which causes an exception after more than six are created — but I don't want to create the AudioContext outside of StreamVisualizer (i.e. create it in main.js and test in `gotRemoteStream`). I'll ask Chris Wilson if there's a way around this problem.

Comment (KaptenJansson) [inline src/content/peerconnection/webaudio-output/js/main.js:209]: Re AudioContext: AFAIK the only way around it is either to reuse the AudioContext (i.e. a global shared state) or close it and release all references to it and let the garbage collector clear it.

https://developer.mozilla.org/en-US/docs/Web/API/AudioContext/close

**Comment by KaptenJansson** [comment]
Wonder what happened with travis on the last commit?

**Comment by samdutton** [comment]
restart ICE error and some ?Node error I think.

https://travis-ci.org/samdutton/webrtc/jobs/106169919

**Comment by KaptenJansson** [comment]
I meant the overall status, it still says waiting for status. Anyway. Will merge as it's not related to this.

**Pull request merged**